master/pkg/model/length.go

```diff
@@ -105,14 +105,14 @@ func NewLengthInEpochs(epochs int) Length {
 }
 
 // UnitsFromBatches return the number of units completed by the given batches, rounded up.
-func UnitsFromBatches(batches int, ctx UnitContext) float64 {
+func UnitsFromBatches(ctx UnitContext, batches int) Length {
 	switch ctx.defaultUnit {
 	case Records:
-		return float64(batches * ctx.globalBatchSize)
+		return Length{Unit: Records, Units: batches * ctx.globalBatchSize}
 	case Batches:
-		return float64(batches)
+		return Length{Unit: Batches, Units: batches}
 	case Epochs:
-		return float64(batches*ctx.globalBatchSize) / float64(ctx.recordsPerEpoch)
+		return Length{Unit: Epochs, Units: (batches * ctx.globalBatchSize) / ctx.recordsPerEpoch}
```

Comment:
Will this result in far less granular progress updates?

Also the comment says "rounded up" but this looks like rounding down to me.

Also rounding up seems like the wrong thing to do anyway.

Reply:
yeah, i realized this while dozing off last night and forgot until now, thanks for reminding me. back to a unit-less float it is.